src/createcompendia/leftover_umls.py

```diff
@@ -0,0 +1,239 @@
+        biolink_toolkit = Toolkit()
+
+        for compendium in compendia:
+            logging.info(f"Starting compendium: {compendium}")
+            umls_ids = set()
+
+            with open(compendium, 'r') as f:
+                for row in f:
+                    cluster = json.loads(row)
+                    for id in cluster['identifiers']:
+                        if id['i'].startswith(UMLS + ':'):
+                            umls_ids.add(id['i'])
+
+            logging.info(f"Completed compendium {compendium} with {len(umls_ids)} UMLS IDs")
+            umls_ids_in_other_compendia.update(umls_ids)
+
+        logging.info(f"Completed all compendia with {len(umls_ids_in_other_compendia)} UMLS IDs.")
+        reportf.write(f"Completed all compendia with {len(umls_ids_in_other_compendia)} UMLS IDs.\n")
+        # print(umls_ids_in_other_compendia)
+
+        # Load all the semantic types.
+        types_by_id = dict()
+        types_by_tui = dict()
+        with open(mrsty, 'r') as inf:
+            for line in inf:
+                x = line.strip().split('|')
+                umls_id = f"{UMLS}:{x[0]}"
+                tui = x[1]
+                # stn = x[2]
+                sty = x[3]
+
+                if umls_id not in types_by_id:
+                    types_by_id[umls_id] = dict()
+                if tui not in types_by_id[umls_id]:
+                    types_by_id[umls_id][tui] = set()
+                types_by_id[umls_id][tui].add(sty)
+
+                if tui not in types_by_tui:
+                    types_by_tui[tui] = set()
+                types_by_tui[tui].add(sty)
+
+        logging.info(f"Completed loading {len(types_by_id.keys())} UMLS IDs from MRSTY.RRF.")
+        reportf.write(f"Completed loading {len(types_by_id.keys())} UMLS IDs from MRSTY.RRF.\n")
+
+        with open('babel_outputs/reports/umls-types.tsv', 'w') as outf:
+            for tui in sorted(types_by_tui.keys()):
+                for sty in sorted(list(types_by_tui[tui])):
+                    outf.write(f"{tui}\t{sty}\n")
+
+        # Create a compendium that consists solely of all MRCONSO entries that haven't been referenced.
+        # Code adapted from datahandlers.umls.build_sets()
+        count_no_umls_type = 0
+        count_multiple_umls_type = 0
+        with open(mrconso, 'r') as inf:
+            for line in inf:
+                x = line.strip().split('|')
+                cui = x[0]
+                umls_id = f"{UMLS}:{cui}"
+                if umls_id in umls_ids_in_other_compendia:
+                    logging.debug(f"UMLS ID {umls_id} is in another compendium, skipping.")
+                    continue
+                if umls_id in umls_ids_in_this_compendium:
+                    logging.debug(f"UMLS ID {umls_id} has already been included in this compendium, skipping.")
+                    continue
+                lang = x[1]
+
+                # TODO: note that this technically skips terms that don't have any English labels.
```

Comment:
Is there a way to have this suppression/language logic in one place?   We have changed it a few times, so I'm worried about it getting fixed in one place and not another.

Reply:
Centralized the suppression/language code in 335c010b733361954.